pkg/hub/addon/addon.go

```diff
@@ -81,13 +91,26 @@ func (a *clusterProxyAddOnAgent) Manifests(cluster *clusterv1.ManagedCluster, ad
 		return nil, fmt.Errorf("no data in ca-bundle-crt configmap: %v", caConfigMap)
 	}
 
-	host, port := a.anpPublicHost, a.anpPublicPort
+	agentImage, err := a.imageRegistryClient.Cluster(cluster.Name).ImageOverride(a.agentImage)
+	if err != nil {
+		return nil, err
+	}
+
+	imagePullSecret := os.Getenv(defaultImagePullSecretEnvVarName)
+	pullSecret, err := a.imageRegistryClient.Cluster(cluster.Name).PullSecret()
+	if err != nil {
+		return nil, err
+	}
+	if pullSecret != nil {
+		imagePullSecret = pullSecret.Name
```

Comment:
pullSecret may not in the same ns with the pod

Reply:
fixed to `open-cluster-management-image-pull-credentials`